rust/capture/src/config.rs

```diff
@@ -58,6 +58,8 @@ pub struct Config {
 
     #[envconfig(default = "events")]
     pub capture_mode: CaptureMode,
+
+    pub concurrency_limit: Option<usize>,
```

Comment:
I think you're better off decorating this with `envconfig` and setting a (high) default

Reply:
i did that originally but then got worried about messing up events unnecessarily - this way i can be very confident nothing breaks if we don't change that env var

what exactly does the envconfig decorator do that would help here?

Reply:
gonna leave it optional for now but happy to change if we want this on events too